src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -10648,202 +10666,1371 @@ static void Main()
+        ) cil managed 
+    {
+        .maxstack 8
+
+        IL_0000: ldnull
+        IL_0001: throw
+    } // end of method B::Equals
+
+    .method family hidebysig specialname rtspecialname 
+        instance void .ctor (
+            class B ''
+        ) cil managed 
+    {
+        .maxstack 8
+
+        IL_0000: ldnull
+        IL_0001: throw
+    } // end of method B::.ctor
+
+    .method public hidebysig specialname rtspecialname 
+        instance void .ctor () cil managed 
+    {
+        .maxstack 8
+
+        IL_0000: ldnull
+        IL_0001: throw
+    } // end of method A::.ctor
+} // end of class B
+
+";
+            var source = @"
+public record C : B {
+}
+
+public record D : B {
+    new protected virtual System.Type EqualityContract
+        => throw null;
+}
+
+public record E : B {
+    new protected virtual int EqualityContract
+        => throw null;
+}
+
+public record F : B {
+    new protected virtual Type EqualityContract
+        => throw null;
+}
+
+public record G : B {
+    protected override System.Type EqualityContract
+        => throw null;
+}
+";
+            var comp = CreateCompilationWithIL(new[] { source, IsExternalInitTypeDefinition }, ilSource: ilSource, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyEmitDiagnostics(
+                // (2,15): error CS8876: 'C.EqualityContract' does not override expected property from 'B'.
+                // public record C : B {
+                Diagnostic(ErrorCode.ERR_DoesNotOverrideBaseEqualityContract, "C").WithArguments("C.EqualityContract", "B").WithLocation(2, 15),
+                // (6,39): error CS8876: 'D.EqualityContract' does not override expected property from 'B'.
+                //     new protected virtual System.Type EqualityContract
+                Diagnostic(ErrorCode.ERR_DoesNotOverrideBaseEqualityContract, "EqualityContract").WithArguments("D.EqualityContract", "B").WithLocation(6, 39),
+                // (11,31): error CS8874: Record member 'E.EqualityContract' must return 'Type'.
+                //     new protected virtual int EqualityContract
+                Diagnostic(ErrorCode.ERR_SignatureMismatchInRecord, "EqualityContract").WithArguments("E.EqualityContract", "System.Type").WithLocation(11, 31),
+                // (16,27): error CS0246: The type or namespace name 'Type' could not be found (are you missing a using directive or an assembly reference?)
+                //     new protected virtual Type EqualityContract
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "Type").WithArguments("Type").WithLocation(16, 27),
+                // (21,36): error CS8876: 'G.EqualityContract' does not override expected property from 'B'.
+                //     protected override System.Type EqualityContract
+                Diagnostic(ErrorCode.ERR_DoesNotOverrideBaseEqualityContract, "EqualityContract").WithArguments("G.EqualityContract", "B").WithLocation(21, 36)
+                );
+        }
```

Comment:
I couldn't find a test verifying IL for EqualityContract implementation. Consider adding one

Reply:
> I couldn't find a test verifying IL for EqualityContract implementation. Consider adding one

I am not changing anything about emitted IL and I assume that that kind of testing has been already done, if necessary.

---
In reply to: [452487751](https://github.com/dotnet/roslyn/pull/45831#discussion_r452487751) [](ancestors = 452487751)

Reply:
I'm saying there doesn't seem to be any (that I could find), so may be a good occasion to cover, while in the area.

---
In reply to: [452488774](https://github.com/dotnet/roslyn/pull/45831#discussion_r452488774) [](ancestors = 452488774,452487751)

Reply:
> I'm saying there doesn't seem to be any (that I could find), so may be a good occasion to cover, while in the area.

Given simplicity of what this property does, I do not believe IL verification is warranted. Behavior tests are good enough. I will add assert in a couple of tests to verify the actual value the property returns. 

---
In reply to: [452489813](https://github.com/dotnet/roslyn/pull/45831#discussion_r452489813) [](ancestors = 452489813,452488774,452487751)